### DISCUSSION TITLE
5576-003 Anbindung funktioniert für eine Stunde, danach Modbus CRC Check failed

### DISCUSSION THREAD

**Discussion by yufaxt** [Tom-Bom-badil/samson_trovis_557x#33] [General (de/en)]
Hallo und vielen Dank für das großartige Projekt!

Es ist mir gelungen, einen ESP32-C3 mit ESPHome direkt an den 5576-003 (V2.13) anzuschließen. Das Kabel ist wie im Wiki beschrieben, etwa 20 cm lang, sodass es wahrscheinlich kein Problem mit der Kabellänge gibt. 5V werden extern über USB zugeführt. Es funktioniert perfekt für etwa eine Stunde, danach tritt ein Fehler auf. Die Logs zeigen Folgendes:

[14:19:27][D][modbus_controller:040]: Modbus command to device=247 register=0x03 countdown=0 no response received - removed from send queue
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 607D!=800
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 73!=12
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 76!=04
[14:19:28][W][modbus:114]: Modbus CRC Check failed! C5D1!=08
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 4A51!=22
[14:19:28][W][modbus:114]: Modbus CRC Check failed! D8F!=200
[14:19:28][W][modbus:114]: Modbus CRC Check failed! C071!=02
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 504C!=101
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 7450!=01
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 7450!=101
[14:19:28][W][modbus:114]: Modbus CRC Check failed! 6DD0!=01
[14:19:29][D][modbus_controller:040]: Modbus command to device=247 register=0x38 countdown=0 no response received - removed from send queue

Hat jemand eine Idee, woran das liegen könnte? Vielen Dank im Voraus!

**Comment by Tom-Bom-badil** [comment]
Die wahrscheinlichste Fehlerquelle liegt aus meiner Sicht in der Software auf dem ESP (z.B. fehlerhafte Buffer-Programmierung oder Speicherfreigaben im verwendeten Modbus-Stack). Was verwendest Du da?

Um das Problem einzukreisen:
Kannst Du bitte mal in TrovisView die Logfunktion anwerfen und prüfen, ob der Fehler dort nach einer Stunde ebenfalls auftritt? Dabei möglichst das gleiche Kabel verwenden. Falls der Fehler dann nicht auftritt, liegt es nahe, dass das Problem am ESP bzw. dessen Software liegt.

/tom

**Reply by yufaxt** [reply]
vielen Dank für die Rückmeldung!

Ich verwende einfach ESPHome. Komischerweise funktioniert alles nicht auch nach einem Neustart. Ich konnte es allerdings mit anderen Pins als TX/RX zum laufen bringen, aber wieder nur für ungefähr eine Stunde. 
Ich werde es nächste Woche mit einem m5stack nanoc6 probieren, mit der Hoffnung, dass es doch irgendwie an meinem ESP32-Board liegt.

Wenn ich alles richtig verstehe, für TrovisView brauche ich noch einen Adapter wie z.B USR-TCP232-E2. Den würde ich dann kaufen, wenn es auch mit dem NanoC6 nicht geht.

**Reply by Tom-Bom-badil** [reply]
Mit ESPHome habe noch keine Erfahrung - allerdings laufen hier abwechselnd ein USR K7 und ein TCP232-E2 in meiner Bastelstrecke wochenlang völlig problemlos mit HA und dessen eingebauten Modbus-Funktionen. Ehrlich gesagt finde ich diese Vorgehensweise auch ein wenig transparenter als ein zusätzliches ESPHome, u.a. für das Debugging. Unter '[Code](https://github.com/Tom-Bom-badil/samson_trovis_557x)' habe ich übrigens begonnen, meine Bastel-Config von HA zu publishen, das sollte beim Start helfen.

Der Vorteil der USR-Adapter ist übrigens, dass Du bei Bedarf auch temporär auf Modbus RTU statt TCP umschalten kannst. Ich konnte zB mit dem Samson Firmware Updater noch nie erfolgreich unter TCP flashen, weder die operative 5576 noch die Bastel-5579; unter RTU mit virtuellem COMx: per VCom war das immer auf Anhieb problemlos möglich. Danach halt wieder Umschaltung auf TCP Regelbetrieb, und gut ist's.

/tom

**Reply by yufaxt** [reply]
Das stimmt. Ich habe gehofft, durch die ESPHome-Lösung auch die LAN-Verkabelung zu vermeiden und dann die USR Option mit provisorischem Kabel nur vorübergehend nutzen, falls ich das brauche (z.B Firmware Upgrade). Apropos, wegen EOL (https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/31), wo kann ich die letzte Version bekommen? Kriegt man das vom Support? 

Vielen Dank!

**Reply by Tom-Bom-badil** [reply]
Uff, Du hast Recht - ich finde die Firmwares jetzt auch nicht mehr. Selbst von den aktuell noch vertriebenen Reglern laufen einige Links in's Leere - - zur Zeit wird wohl die Webseite überarbeitet.

Ich konnte aber noch die 5573-5579_V263.B55 [hier](https://leusch.samsongroup.com/de/produkte/automationssysteme/5578/) finden, auch wenn die Seite für ein anderes Modell ist (rechte Maustaste, speichern unter auf dem Link für die Firmware). V 2.63 ist meines Wissens der letzte Stand für die 5576 und 5579. Hoffe, das hilft Dir weiter ...

/tom

**Reply by yufaxt** [reply]
Letzter Stand:

Mit dem M5Stack Nano C6 hatte ich leider die gleichen Probleme, egal ob mit externer Stromversorgung oder über den Trovis.

Beim TCP232-E2 lief anfangs alles ohne Probleme. Ich konnte sogar die Firmware auf Version 2.63 aktualisieren. Nach ein paar Tagen jedoch war die Stromversorgung über den Trovis nicht mehr ausreichend – es wurden nur noch ca. 4V gemessen. Das E2 schaltete ständig ein und aus.

Mit einer externen Stromversorgung läuft alles jedoch sehr stabil. Vielen Dank!
